/*
    Copyright 2020 Google LLC

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto2";
package hologramAvailability;

message HologramDataAvailability {
  // Listing of different data sources
  enum DataType {
     PLAY_COUNTRY = 0;
     APP_COUNTRY_PUBLISH_TIME = 1;
     SPAM = 2;
     QUERY_CATEGORY_SOURCE = 3;
     APPS_DAILY_DATA_SOURCE = 4;
     UNIFIED_USER_DATA_SOURCE = 5;
     TEST_CODE = 6;
     DEVICE_INTEGRITY_SOURCE = 7; 
     SHIELD_ID = 8;
     SERVER_EXPERIMENT = 9; 
     APPS_TITLE_TIER = 10;
     APP_TYPE_SOURCE = 11;
     ANDROID_SPAM_ABUSE = 12;
     USER_EXPERIMENT_ID_AND_TEST_CODE = 13;
     USER_ATTRIBUTE = 14;
     PRIME_MERIDIAN= 15;
  }

  // Details about the status and whatâ€™s causing errors
  enum StatusType {
      UNKNOWN = 0;
      SUCCESS = 1;
      DATA_NOT_INJECTED = 2;
  }

  // Specification about which data source is currently accessed
  optional DataType data_source = 1;

  // Specification about the status of the data source
  message DataSourceDetail {
      optional int64 date = 1;
      optional StatusType status = 2;
  }
  
  // Current day status
  optional DataSourceDetail latest_status = 2;

  // An array of status of up to the last 7 days.
  repeated DataSourceDetail history = 3;
}